src/components/ContactsList/ContactsListItem.vue

```diff
@@ -85,6 +112,26 @@ export default {
 				this.avatarUrl = `${this.source.url}?photo`
 			}
 		},
+
+		isSelected() {
+			this.ischecked = true
+			// update the selected
+			this.$parent.$parent.$emit('update-check-selected', this.source.key)
+		},
+
+		isUnselected() {
+			this.ischecked = false
+			// update the selected
+			this.$parent.$parent.$emit('update-check-selected', this.source.key)
```

Comment:
accessing the grandparent is a no go. this makes the component depend on the exact structure and is prone to break in unnoticed ways. Please find a better way to pass around the data. Could you use Vuex?